packages/__docs__/src/NavToggle/index.tsx

```diff
@@ -23,32 +23,29 @@
  */
 
 import React, { Component } from 'react'
-import PropTypes from 'prop-types'
 
 import { ToggleDetails } from '@instructure/ui-toggle-details'
 import { Text } from '@instructure/ui-text'
 import { View } from '@instructure/ui-view'
 import { InstUISettingsProvider } from '@instructure/emotion'
 import { instructure } from '@instructure/ui-themes'
-
-class NavToggle extends Component {
-  static propTypes = {
-    summary: PropTypes.string.isRequired,
-    variant: PropTypes.oneOf(['section', 'category']),
-    children: PropTypes.node
-  }
-
+import type { NavToggleProps } from './props'
+import { propTypes, allowedProps } from './props'
+class NavToggle extends Component<NavToggleProps> {
+  static propTypes = propTypes
+  static allowedProps = allowedProps
   static defaultProps = {
     variant: 'section',
     children: null
   }
+  private _toggle: any
```

Comment:
You can tell its type from the ref: 
```suggestion
  private _toggle: ToggleDetails | null = null
```

And in the render method:

```js
       <ToggleDetails
          ref={(c: ToggleDetails) => {
            this._toggle = c
          }}
          fluidWidth
          {...props}
          summary={summaryContent}
        />
```

Reply:
can you explain about `null = null` part?

Reply:
So the type of _toggle is **either** `ToggleDetails` **or** `null`:
```js
private _toggle: ToggleDetails | null
```

But because it is either one of these types, it cannot be `undefined`, and it is undefined if you don't give it an initial value. 
```js
private _toggle // = undefined
```

And since it cannot initially be `ToggleDetails` type, we set it to null
```js
private _toggle: ToggleDetails | null = null
```